libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/car/context/MapboxCarContextServiceDelegate.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.androidauto.internal.car.context
+
+import androidx.car.app.CarContext
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.androidauto.car.MapboxCarContext
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+/**
+ * Delegate function for the [MapboxCarContextServiceDelegate]. This gives you the ability to
+ * scope a service to the [CarContext] lifecycle. If the service is accessed when the [CarContext]
+ * is not available, it throw an [IllegalStateException].
+ */
+fun <T> MapboxCarContext.mapboxCarService(
+    name: String,
+    factory: () -> T
+): ReadOnlyProperty<Any, T> = MapboxCarContextServiceDelegate(this, name, factory)
+
+class MapboxCarContextServiceDelegate<T>(
+    val mapboxCarContext: MapboxCarContext,
+    val name: String,
```

Comment:
Seems to be unused.

Reply:
Hmm yeah i had it for error reporting but lost it when refactoring. Added one back in.